noteit/src/main/java/com/noteit/service/UserServiceImpl.java

```diff
@@ -35,8 +36,15 @@ public User findByUsername(String username) {
     }
 
     @Override
-    public User save(User theUser) {
-        return userRepository.save(theUser);
+    public User save(User theUser) throws UserNameAlreadyExistsException {
+        try {
+            findByUsername(theUser.getUsername());
+
+        } catch (UsernameNotFoundException e){
+            return userRepository.save(theUser);
+        }
+

```

Comment:
This implementation is not wrong, but it could cause readers feel awkward. Normally the main goal of method, which is saving `User` in this case, should not put in the `catch` clause. Reader usually expects things inside `catch` are to handle unexpected problems.
The flow of implementation should be checking whether an user with input username exist. If yes, throws exception otherwise saving the user.